src/uc_paymentwall.module

```diff
@@ -2,308 +2,324 @@
+                '#description' => t('All data your can get from <a href="https://www.paymentwall.com/en/documentation/Ubercart/831">https://www.paymentwall.com/en/documentation/Ubercart/831</a><br>'),
+            );
+
+            $form['uc_pw_app_key'] = array(
+                '#type' => 'textfield',
+                '#title' => t('Project key:'),
+                '#description' => t('Your Project key.'),
+                '#default_value' => variable_get('uc_pw_app_key', ''),
+            );
+
+            $form['uc_pw_secret_key'] = array(
+                '#type' => 'textfield',
+                '#title' => t('Secret key:'),
+                '#description' => t('Your secret key.'),
+                '#default_value' => variable_get('uc_pw_secret_key', ''),
+            );
+
+            $form['uc_pw_widget_code'] = array(
+                '#type' => 'textfield',
+                '#title' => t('Widget code:'),
+                '#description' => t('Your widget code.'),
+                '#default_value' => variable_get('uc_pw_widget_code', ''),
+            );
+
+            $form['uc_pw_success_url'] = array(
+                '#type' => 'textfield',
+                '#title' => t('Success url:'),
+                '#description' => t('Success url.'),
+                '#default_value' => variable_get('uc_pw_success_url', ''),
+            );
+
+            $form['uc_pw_test_mode'] = array(
+                '#type' => 'checkbox',
+                '#title' => t('Enable test mode'),
+                '#default_value' => variable_get('uc_pw_test_mode', FALSE),
+            );
+
+            $form['uc_pw_delivery'] = array(
+                '#type' => 'checkbox',
+                '#title' => t('Enable Delivery Confirmation API'),
+                '#default_value' => variable_get('uc_pw_delivery', TRUE),
+            );
+
+            $form['actions']['submit'] = array(
+                '#type' => 'submit',
+                '#value' => t('Save'),
+            );
+
+            return $form;
     }
-  } else {
-    drupal_set_message(t('Fill all fields'), 'error');
-  }
-
-  drupal_goto('admin/store/settings/payment/method/paymentwall');
-  die();
 }
 
 /**
  * Get order data and redirect to Paymentwall widget page.
  */
-function uc_paymentwall_form($form, &$form_state, $order) {
-  $_SESSION['order'] = base64_encode(serialize($order));
+function uc_paymentwall_form($form, &$form_state, $order)
+{
+    $_SESSION['order'] = base64_encode(serialize($order));
 
-  $form['#action'] = url('paymentwall_complete');
-  $form['actions']['submit'] = array(
-    '#type' => 'submit',
-    '#value' => t('Submit Orders'),
-  );
+    $form['#action'] = url('paymentwall_complete');
+    $form['actions']['submit'] = array(
+        '#type' => 'submit',
+        '#value' => t('Submit Orders'),
+    );
 
-  return $form;
+    return $form;
 }
 
 /**
- * @param $data
- * Init Paymentwall config
+ * Complate order on example.com/paymentwall_complete
```

Comment:
it's a bit wrong comment, because user can install ubercart into example.com/shop or anywhere else